lib/kubeclient/aws_eks_credentials.rb

```diff
@@ -20,18 +20,26 @@ def token(credentials, eks_cluster, region: 'us-east-1')
         end
         # https://github.com/aws/aws-sdk-ruby/pull/1848
         # Get a signer
-        signer = Aws::Sigv4::Signer.new(
-          service: 'sts',
-          region: region,
-          credentials: credentials
-        )
+        if is_credentials_provider
```

Comment:
From `Aws::Sigv4::Signer`:

> A credential provider is any object that responds to `#credentials`


This got me thinking, what if instead of adding another parameter we did
```suggestion
        if credentials.respond_to?(:credentials)
```
just to make it a little easier on the end user.
Unfortunately AWS use modules rather than a `Aws::CredentialProvider` base class so we can't check `kind_of?`.

Reply:
Nice Suggestion. Modified and committed the suggested changes and updated the readme file accordingly.